test/ssc_test/cmod_battery_pvsamv1_test.cpp

```diff
@@ -0,0 +1,448 @@
+	ssc_data_set_matrix(data, "ur_ec_sched_weekend", p_ur_ec_sched_weekend_srp, 12, 24);
+	ssc_number_t p_ur_ec_tou_mat_srp[36] = { 1, 1, 9.9999999999999998e+37, 0, 0.2969, 0, 2, 1, 9.9999999999999998e+37, 0, 0.081900000000000001, 0, 3, 1, 9.9999999999999998e+37, 0, 0.34989999999999999, 0, 4, 1, 9.9999999999999998e+37, 0, 0.083599999999999994, 0, 5, 1, 9.9999999999999998e+37, 0, 0.123, 0, 6, 1, 9.9999999999999998e+37, 0, 0.074999999999999997, 0 };
+	ssc_data_set_matrix(data, "ur_ec_tou_mat", p_ur_ec_tou_mat_srp, 6, 6);
+
+	int pvsam_errors = run_pvsam1_battery_ppa(data);
+	EXPECT_FALSE(pvsam_errors);
+
+	if (!pvsam_errors)
+	{
+		ssc_number_t annual_energy;
+		ssc_data_get_number(data, "annual_energy", &annual_energy);
+		EXPECT_NEAR(annual_energy, expectedEnergy, m_error_tolerance_hi) << "Annual energy.";
+
+		auto data_vtab = static_cast<var_table*>(data);
+		auto annualChargeEnergy = data_vtab->as_vector_ssc_number_t("batt_annual_charge_energy");
+		EXPECT_NEAR(annualChargeEnergy[1], expectedBatteryChargeEnergy, m_error_tolerance_hi) << "Battery annual charge energy.";
+
+		auto annualDischargeEnergy = data_vtab->as_vector_ssc_number_t("batt_annual_discharge_energy");
+		EXPECT_NEAR(annualDischargeEnergy[1], expectedBatteryDischargeEnergy, m_error_tolerance_hi) << "Battery annual discharge energy.";
+
+		auto batt_power = data_vtab->as_vector_ssc_number_t("batt_power");
+		daily_battery_stats batt_stats = daily_battery_stats(batt_power);
+
+		EXPECT_NEAR(batt_stats.peakKwCharge, peakKwCharge, m_error_tolerance_lo);
+		EXPECT_NEAR(batt_stats.peakKwDischarge, peakKwDischarge, m_error_tolerance_lo);
+		EXPECT_NEAR(batt_stats.peakCycles, peakCycles, m_error_tolerance_lo);
+		EXPECT_NEAR(batt_stats.avgCycles, avgCycles, 0.0001); // Runs once per year
+	}
+}
+
+/// Test PVSAMv1 with all defaults and DC battery enabled with custom dispatch and PPA financial model
+TEST_F(CMPvsamv1BatteryIntegration_cmod_pvsamv1, PPA_CustomDispatchBatteryModelDCIntegration)
+{
+	ssc_data_t data = ssc_data_create();
+	pvsamv1_pv_defaults(data);
+	pvsamv1_battery_defaults(data);
+	grid_and_rate_defaults(data);
+	singleowner_defaults(data);
+
+	ssc_number_t expectedEnergy = 37321520;
+	ssc_number_t expectedBatteryChargeEnergy = 2030;
+	ssc_number_t expectedBatteryDischargeEnergy = 3175;
+
+	ssc_number_t peakKwCharge = -1020.4;
+	ssc_number_t peakKwDischarge = 958.7;
+	ssc_number_t peakCycles = 1;
+	ssc_number_t avgCycles = 0.0027;
+
```

Comment:
[custom_dispatch_singleowner_hourly.csv.zip](https://github.com/NREL/ssc/files/4383925/custom_dispatch_singleowner_hourly.csv.zip)

Here's another custom dispatch you can try. Again, not perfect but gives a bit more signal to test